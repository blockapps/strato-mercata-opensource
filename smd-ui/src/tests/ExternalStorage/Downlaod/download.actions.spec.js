import {
  openDownloadModal,
  closeDownloadModal,
  downloadRequest,
  downloadSuccess,
  downloadFailure,
  resetError,
  clearUrl
} from "../../../components/ExternalStorage/Download/download.actions";
import { error } from "../storageMock";

describe('Download: actions', () => {

  describe('Modal:', () => {

    test('open', () => {
      expect(openDownloadModal()).toMatchSnapshot();
    });

    test('close', () => {
      expect(closeDownloadModal()).toMatchSnapshot();
    });

  });

  describe('Downlaod:', () => {

    test('request', () => {
      const contractAddress = 'c918420c68346af5fe2aef067faf7b103afde5ed';
      expect(downloadRequest(contractAddress)).toMatchSnapshot();
    });

    test('success', () => {
      const url = 'https://strato-external-storage.s3.amazonaws.com/1529915329415-widescreen.jpeg';
      expect(downloadSuccess(url)).toMatchSnapshot();
    });

    test('failure', () => {
      expect(downloadFailure(error)).toMatchSnapshot();
    });

  });

  test('reset error', () => {
    expect(resetError()).toMatchSnapshot();
  });

  test('clear Url', () => {
    expect(clearUrl()).toMatchSnapshot();
  });

});
