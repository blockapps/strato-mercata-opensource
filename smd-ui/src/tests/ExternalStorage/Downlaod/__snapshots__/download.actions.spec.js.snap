// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Download: actions Downlaod: failure 1`] = `
Object {
  "error": "ERROR",
  "type": "DOWNLOAD_FAILURE",
}
`;

exports[`Download: actions Downlaod: request 1`] = `
Object {
  "contractAddress": "c918420c68346af5fe2aef067faf7b103afde5ed",
  "type": "DOWNLOAD_REQUEST",
}
`;

exports[`Download: actions Downlaod: success 1`] = `
Object {
  "type": "DOWNLOAD_SUCCESS",
  "url": "https://strato-external-storage.s3.amazonaws.com/1529915329415-widescreen.jpeg",
}
`;

exports[`Download: actions Modal: close 1`] = `
Object {
  "type": "CLOSE_DOWNLOAD_MODAL",
}
`;

exports[`Download: actions Modal: open 1`] = `
Object {
  "type": "OPEN_DOWNLOAD_MODAL",
}
`;

exports[`Download: actions clear Url 1`] = `
Object {
  "type": "CLEAR_URL",
}
`;

exports[`Download: actions reset error 1`] = `
Object {
  "type": "RESET_ERROR",
}
`;
